### PR TITLE
Replace NiPype's filemanip.py

### PR DESCRIPTION
**Pull request by madisoth**
## Changes proposed in this pull request

Replaces NiPype's filemanip.py with a modified version, which adds CIFTI file extensions to the built-in list of "special extensions". 

"Special extensions" are multi-part extensions (e.g. ".nii.gz", ".dtseries.nii") which NiPype is directed to treat as a single extension for filemanip functions.

This fixes an issue in 0.1.0 where using --cifti without --despike would error, due to the censored BOLD from CensorScrub having a bad file extension of ".dtseries_bold.nii". Also fixes other cases of CIFTI files in the work directory with bad extensions. 

Also corrected a misindented try-except-else in filemanip.py (L746)

### CODE REVIEW

File: xcp_d/interfaces/connectivity.py [hunk 1]
```diff
@@ -112,28 +120,31 @@ def get_atlas_nifti(atlasname):
...
+    elif atlasname == "tiansubcortical":
         atlasfile = pkgrf(
-            'xcp_d',
-            'data//niftiatlas/TianSubcortical/Tian_Subcortex_S3_3T.nii.gz')
+            "xcp_d", "data//niftiatlas/TianSubcortical/Tian_Subcortex_S3_3T.nii.gz"
+        )
     else:
-        raise RuntimeError('atlas not available')
+        raise RuntimeError("atlas not available")
     return atlasfile
```
Comment (mattcieslak) [inline xcp_d/interfaces/connectivity.py:148]: Note to self: this should get a refactor to be more readable

File: xcp_d/interfaces/prepostcleaning.py [hunk 2]
```diff
@@ -67,48 +79,52 @@ def _run_interface(self, runtime):
         # Check if we need to do anything
         if self.inputs.initial_volumes_to_drop == 0:
             # write the output out
-            self._results['bold_file_dropped_TR'] = self.inputs.bold_file
-            self._results['fmriprep_confounds_file_dropped'
-                          '_TR'] = self.inputs.fmriprep_confounds_file
+            self._results["bold_file_dropped_TR"] = self.inputs.bold_file
+            self._results[
+                "fmriprep_confounds_file_dropped" "_TR"
```
Comment (mattcieslak) [inline xcp_d/interfaces/prepostcleaning.py:84]: This is an odd way to do it. Did you also run a ton of auto-linting for this PR?

File: xcp_d/utils/filemanip.py [hunk 3]
```diff
@@ -0,0 +1,985 @@
...
+        ".pscalar.nii",
+        ".ptseries.nii",
+        ".sdseries.nii",
+        ".label.gii",
+        ".label.gii",
+        ".func.gii",
+        ".shape.gii",
+        ".rgba.gii",
+        ".surf.gii",
+    ]
```
Comment (mattcieslak) [inline xcp_d/utils/filemanip.py:114]: nice, this would be worth doing a PR on nipype at some point too

File: xcp_d/utils/filemanip.py [hunk 4]
```diff
@@ -0,0 +1,985 @@
...
+
+
+@contextlib.contextmanager
+def indirectory(path):
+    cwd = os.getcwd()
+    os.chdir(str(path))
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
```
Comment (mattcieslak) [inline xcp_d/utils/filemanip.py:985]: do we need all of the other functions in here? Or just the one that supports the cifti extensions?

**Review by mattcieslak — COMMENTED** [review]
In the future could you do separate PRs for linting-related changes and workflow-related changes? I assumed that only filemanip.py and files that now import from it are the ones with content changes

**Comment by madisoth** [comment]
@mattcieslak agree that this PR should be cleaned up a bit -- I'll convert this PR to a draft for now and resubmit later

**Review by mattcieslak — APPROVED** [review]
Excellent!!

**Pull request merged**